Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.cs

```diff
@@ -55,26 +55,24 @@ public override IEnumerable<IPortfolioTarget> CreateTargets(QCAlgorithmFramework
                 return targets;
             }
 
-            // Get symbols that have emit insights, are still in the universe, and insigths haven't expired
-            var symbols = _insightCollection
-                .Where(x => x.CloseTimeUtc > algorithm.UtcTime)
-                .Select(x => x.Symbol).Distinct().ToList();
+            // Get last insight that haven't expired of each symbol that is still in the universe
+            var activeInsights = (from insight in _insightCollection
+                                  group insight by insight.Symbol into g
+                                  where g.LastOrDefault().CloseTimeUtc > algorithm.UtcTime
+                                  select g.LastOrDefault()).ToList();
```

Comment:
We could be losing insights here. Maybe a better option is to sort by generated time and then grabbing the 'last' isn't arbitrary -- we would simply be grabbing the most recent insight.